codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java

```diff
@@ -354,4 +284,38 @@ public ByteString path() {
     public ByteString status() {
         return get(PseudoHeaderName.STATUS.value());
     }
+
+    @Override
+    protected final HeaderEntry newHeaderEntry(int h, ByteString name, ByteString value, HeaderEntry next) {
+        return new Http2HeaderEntry(h, name, value, next);
+    }
+
+    private final class Http2HeaderEntry extends HeaderEntry {
+        protected Http2HeaderEntry(int hash, ByteString key, ByteString value, HeaderEntry next) {
+            super(hash, key);
+            this.value = value;
+            this.next = next;
+
+            // Make sure the pseudo headers fields are first in iteration order
+            if (!key.isEmpty() && key.byteAt(0) == ':') {
```

Comment:
@buchgr - Yes. Building on your PR means that I inherited your unit tests :)

DefaultHttp2HeadersTest .pseudoHeadersMustComeFirstWhenIterating